src/SpanInterface.php

```diff
@@ -12,23 +14,24 @@ interface SpanInterface
 {
     /**
      * Sets a single attribute on the span.
-     * NOTE: not sure about "mixed" type hint on $value parameter.
-     * Should it be scalar|Stringable ? An array is acceptable here?
+     * NOTE: value MUST be a scalar (or stringable) value, nesting is supported via key "dot notation"
      *
      * @return $this
      */
-    public function setAttribute(string $key, mixed $value): SpanInterface;
+    public function setAttribute(string $key, string|int|float|bool|Stringable $value): SpanInterface;
 
     /**
      * Sets multiple attributes on this span.
      *
+     * @param iterable<string, string|int|float|bool|Stringable> $attributes
```

Comment:
On the fence as to whether this is correct.
Wondering what the implication of passing an iterable over a KV array would be for providers